### PR TITLE
Queries 1 and 4 complete

### PR DESCRIPTION
**Pull request by aradu12**
I have finished question 1 (#2) and question 4 (#5). 

- added query to fetch all authors session numbers
- added code to task 4 to allow the user to choose between two options (bar plot and individual number), validate their choice and display the appropriate answer.
- added some tweaks and comments to Q1 and Q4 to make it look nicer (page numbers, clearer error msg, more newlines)

**_They are all working on my machine, as well as when sshing to the lab machines. However, here are some notes:_**

- I fixed the NoneType error in Q1 by changing from the pandas API to a cursor. See this [comment](https://github.com/cmput291-wi19-team29/Assignment3/commit/7b4ae4df80edb767e2f12414dda1d276f092a852#commitcomment-32630227) for details

- The TAs have said that in Q4, we should NOT be counting distinct sessions, but ALL sessions. See this eclass [discussion](https://eclass.srv.ualberta.ca/mod/forum/discuss.php?d=1140073#p3001027). 
As such, I'm thinking I should change the display message to say the author is presenting x number of _papers_, instead of participating in x _number of sessions_. Let me know if you think this is a good idea, or if it would confuse the TAs.

- matplotlib graphs are not showing up when sshing to the lab machines unless the DISPLAY env variable is set. See this eclass [question](https://eclass.srv.ualberta.ca/mod/forum/discuss.php?d=1144037#p3011736), which does not yet have a response. This means Q4 and Q6 are not currently showing graphs unless run locally. I suppose the TAs will ensure this var is set when testing our code.

Please let me know what other complaints/concerns/questions you have so I can fix them.
I can see there are some merge conflicts, which I will clean up ASAP and consult u guys on.

### CODE REVIEW

**Comment by aradu12** [comment]
merge conflicts fixed! There were just 2 differences between code I wrote on my branch and code I wrote on the master branch. All the conflicts were within my code, so I didn't change anything else!

File: src/main.py [hunk 1]
```diff
@@ -150,7 +168,72 @@ def task3(conn):
...
+    # read and process SQL query -- get number of sessions per author
+    # note that according to the eclass forum, the session count includes ALL sessions,
+    # not just distinct ones
+    # discussion: https://eclass.srv.ualberta.ca/mod/forum/discuss.php?d=1140073#p3001027
+    
+    query = open('4a.sql','r')
+    sql = query.read()
+    query.close()
+
+    df = None
```
Comment (NinjaPerson24119) [inline src/main.py]: Not C++. df persists beyond the try block so this predeclaration is unnecessary

File: src/main.py [hunk 2]
```diff
@@ -64,17 +70,28 @@ def task1(conn):
...
+      if option.lower() == "n":
+        if showing < batches:
+            # next page
+            showing += 1
+            clear()
+            continue
+        else:
+          # no next page
+          print("\nThere are no more pages. Enter P to go back a page, or choose a paper number.")
+          continue
```
Comment (NinjaPerson24119) [inline src/main.py:85]: Why do you not clear in the event that the page goes out of bounds? Visually, not doing so can defeat the purpose of having the clear at all.

File: src/main.py [hunk 3]
```diff
@@ -64,17 +70,28 @@ def task1(conn):
...
-        continue
+      if option.lower() == "p":
+        if showing >1:
+          # prev page
+          showing -= 1
+          clear()
+          continue
+        else:
+          # no prev page
+          print("\nThere are no previous pages. Enter N to go to the next page, or choose a paper number.")
```
Comment (NinjaPerson24119) [inline src/main.py:95]: Why do you not clear in the event that the page goes out of bounds? Visually, not doing so can defeat the purpose of having the clear at all.

**Comment by NinjaPerson24119** [comment]
- Doesn't clear screen when changing pages out of bounds (is this intentional?)
- Unnecessary predeclaration of df before try block
- Missing descriptive task names in API

Otherwise lgtm. I think it's important to get this merged so that Hugh can push without conflicts, so I'm approving.

**Review by NinjaPerson24119 — APPROVED** [review]

**Comment by NinjaPerson24119** [comment]
I don't think we should worry about the ENV since it's not in our control.
The API is a bit misleading right now because of the spec changes you introduced. I'd add some clarifying info to the API in parenthesis on top of what currently exists

File: src/main.py (hunk 3 above)
Comment (aradu12) [reply src/main.py:95]: Yes, it's intentional. Clearing the screen would make the error message disappear too fast to see. How about I add a time delay using time.sleep, and then clear the screen?

File: src/main.py (hunk 1 above)
Comment (aradu12) [reply src/main.py]: Yeah, I know python doesn't need predecs, just wanted to make it easier for the TAs to follow. I'll remove it.

**Comment by NinjaPerson24119** [comment]
That's a good point. I wouldn't bother changing it.

File: src/main.py (hunk 3 above)
Comment (aradu12) [reply src/main.py:95]: Actually, I think it's better to change it to the delay; it looks a lot nicer.

**Comment by aradu12** [comment]
Merging now so to make Hugh's life easier. The time delay may be a little long (2.5 sec), but for a user that's never seen the errors before, it may take that long for them to digest it. Feel free to change.

**Comment by NinjaPerson24119** [comment]
I think it's fine. Nice solution.

**Pull request merged**